### ISSUE TITLE
Bug: [no-array-constructor] Does not detect `Array` call with optional chaining

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#10933]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.8.2&fileType=.ts&code=IIJxEME8AoEoG4BQjQUgfgHR3kA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQHYHsBaAQ2mmIE9CBjfXZAFzmof2nSnwDMvIAacNkgABBhQAOKatACW4hoRTwZuBgHoCJMpRp1GzVuwyREZNpEEBfEJaA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
Array();

Array?.();
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "no-array-constructor": "off",
    "@typescript-eslint/no-array-constructor": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

`Array?.()` should be reported the same way `Array()` is.

### Actual Result

No report.

### Additional Info

`Array?.()` is reported in the core rule: https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tYXJyYXktY29uc3RydWN0b3I6IFwiZXJyb3JcIiovXG5cbkFycmF5KCk7XG5cbkFycmF5Py4oKTsiLCJvcHRpb25zIjp7InJ1bGVzIjp7fSwibGFuZ3VhZ2VPcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=

💖

**Comment by developer-bandi** [comment]
@JoshuaKGoldberg in our repo's no-array-constructor test file, similer cases are in valid case

```
'Array?.(0, 1, 2);',
'Array?.(x, y);',
```

Since both cases are throwing errors in upstream, should they be fixed the same?

Then logically, new Array, Array, Array? should all be the same.

Is this direction of fixing correct?

https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tYXJyYXktY29uc3RydWN0b3I6IFwiZXJyb3JcIiovXG5cbkFycmF5Py4oMCwgMSwgMik7XG5BcnJheT8uKHgsIHkpOyIsIm9wdGlvbnMiOnsicnVsZXMiOnt9LCJsYW5ndWFnZU9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9fX19fQ==

**Comment by JoshuaKGoldberg** [comment]
Good spot, yup. Anything with optional chaining detected by the core rule but not in this extension rule should be.
